### PR TITLE
Feature/add pagination to left data column

### PR DESCRIPTION
**Pull request by piotrNBTN**

### CODE REVIEW

File: src/components/Pagination/PaginationButton/PaginationButton.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { FC } from "react";
+import { useLanguage } from "@/context/LocaleProvider";
+import { PaginationButtonProps } from "./types";
+import { StyledButton, StyledIconWrapper, StyledText, StyledWrapper } from "./styles";
+
+const PaginationButton: FC<PaginationButtonProps> = ({
+  intent,
+  onLoadNext,
```
Comment (mGasiorek998) [inline src/components/Pagination/PaginationButton/PaginationButton.tsx:8]: I don't think we need two separate functions, just onClick will be sufficient here

File: src/components/Pagination/PaginationButton/PaginationButton.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  icon,
+  isVisible
+}) => {
+  const { loadNext, loadPrevious } = useLanguage();
+  const renderButton = () => {
+    switch (intent) {
+      case "next":
+        return (
+          <StyledWrapper isVisible={isVisible}>
+            <StyledButton onClick={onLoadNext}>
```
Comment (mGasiorek998) [inline src/components/Pagination/PaginationButton/PaginationButton.tsx:19]: onClick={onClick}

File: src/components/Pagination/PaginationButton/PaginationButton.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+              {icon && <StyledIconWrapper>{icon}</StyledIconWrapper>}
+              <StyledText>{loadNext}</StyledText>
+            </StyledButton>
+          </StyledWrapper>
+        );
+      case "previous": {
+        return (
+          <StyledWrapper isVisible={isVisible}>
+            <StyledButton onClick={onLoadPrevious}>
```
Comment (mGasiorek998) [inline src/components/Pagination/PaginationButton/PaginationButton.tsx:28]: Same here

File: src/hooks/types.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import { PaginatedSchedulerData, SchedulerProjectData } from "@/types/global";
+
+export type UsePaginationData = {
+  /**
```
Comment (mGasiorek998) [inline src/hooks/types.ts:4]: Beautiful 👍

File: src/utils/splitToPages.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+          label: data[i + leftIndex].label,
+          data: projects
+        };
+        singlePage.push(newItem);
+        if (i === projectsPerPerson.length - leftIndex - 1) pages.push(singlePage);
+      });
+    }
+
+    return pages;
+  } else {
```
Comment (mGasiorek998) [inline src/utils/splitToPages.ts:49]: Because you have return in line above, you can omit this else

**Review by mGasiorek998 — CHANGES_REQUESTED** [review]

File: src/components/Pagination/PaginationButton/PaginationButton.tsx (hunk 1 above)
Comment (piotrNBTN) [reply src/components/Pagination/PaginationButton/PaginationButton.tsx:8]: correct

**Review by mGasiorek998 — APPROVED** [review]

File: src/components/Pagination/Pagination.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    <>
+      <PaginationButton
+        intent={"previous"}
+        icon={<Icon iconName={"arrowUp"} width="16" height="16" />}
+        onClick={onLoadPrevious}
+        isVisible={pageNum !== 0}
+      />
+      {children && children}
+      <PaginationButton
+        icon={<Icon iconName="arrowDown" width="16" height="16" />}
```
Comment (itlamb) [inline src/components/Pagination/Pagination.tsx:23]: We have two different approaches to setting iconName here - one with `{}` and here without them - please unify

File: src/components/Pagination/Pagination.tsx [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        intent={"next"}
```
Comment (itlamb) [inline src/components/Pagination/Pagination.tsx:24]: Maybe remove all {} from string props here?

File: src/components/Pagination/PaginationButton/PaginationButton.tsx [hunk 8]
```diff
@@ -0,0 +1,36 @@
+import { FC } from "react";
+import { useLanguage } from "@/context/LocaleProvider";
+import { PaginationButtonProps } from "./types";
+import { StyledButton, StyledIconWrapper, StyledText, StyledWrapper } from "./styles";
+
+const PaginationButton: FC<PaginationButtonProps> = ({ intent, onClick, icon, isVisible }) => {
```
Comment (itlamb) [inline src/components/Pagination/PaginationButton/PaginationButton.tsx:6]: I'm not sure about structure of this component - we have 30 lines here and the only difference between variants is the label. Am I missing something?

File: src/hooks/types.ts (hunk 4 above)
Comment (itlamb) [reply src/hooks/types.ts:4]: ❤️

**Review by itlamb — CHANGES_REQUESTED** [review]

**Review by itlamb — APPROVED** [review]

**Pull request merged**